crates/router/src/connector/prophetpay/transformers.rs

```diff
@@ -517,20 +605,75 @@ impl TryFrom<types::RefundsResponseRouterData<api::Execute, ProphetpayRefundResp
     fn try_from(
         item: types::RefundsResponseRouterData<api::Execute, ProphetpayRefundResponse>,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            response: Ok(types::RefundsResponseData {
-                // no refund id is generated, rather transaction id is used for referring to status in refund also
-                connector_refund_id: item.data.request.connector_transaction_id.clone(),
-                refund_status: enums::RefundStatus::from(item.response.response_text),
-            }),
-            ..item.data
-        })
+        if item.response.success {
+            Ok(Self {
+                response: Ok(types::RefundsResponseData {
+                    // no refund id is generated, tranSeqNumber is kept for future usage
+                    connector_refund_id: item.response.tran_seq_number,
+                    refund_status: enums::RefundStatus::Success,
+                }),
+                ..item.data
+            })
+        } else {
+            Ok(Self {
+                status: enums::AttemptStatus::Failure,
+                response: Err(types::ErrorResponse {
+                    code: item.response.response_code,
+                    message: item.response.response_text.clone(),
+                    reason: Some(item.response.response_text),
+                    status_code: item.http_code,
+                    attempt_status: None,
+                }),
+                ..item.data
+            })
+        }
     }
 }
 
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProphetpayRefundSyncResponse {
+    pub success: bool,
+    pub response_text: String,
+    pub response_code: String,
+}
+
+impl<T> TryFrom<types::RefundsResponseRouterData<T, ProphetpayRefundSyncResponse>>
+    for types::RefundsRouterData<T>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<T, ProphetpayRefundSyncResponse>,
+    ) -> Result<Self, Self::Error> {
+        if item.response.success {
+            Ok(Self {
+                response: Ok(types::RefundsResponseData {
+                    // no refund id is generated, rather transaction id is used for referring to status in refund also
+                    connector_refund_id: item.data.request.connector_transaction_id.clone(),
```

Comment:
This will overwrite the tran_seq_number which we are setting in the refund flow. Keep the connector_refund_id here. Take this in separate PR